src/backend/dx11/src/lib.rs

```diff
@@ -1978,17 +2067,19 @@ impl hal::DescriptorPool<Backend> for DescriptorPool {
         let len = layout.handle_count.max(1) as _;
 
         self.allocator.allocate_range(len).map(|range| {
-			for i in range.clone() {
-				self.handles[i] = Descriptor(ptr::null_mut());
-			}
+            for i in range.clone() {
+                self.handles[i] = Descriptor(ptr::null_mut());
+            }
 
             DescriptorSet {
                 offset: range.start,
                 len,
                 handles: unsafe { self.handles.as_mut_ptr().offset(range.start as _) },
                 register_remap: layout.register_remap.clone(),
-                flush_coherent_buffers: RefCell::new(Vec::new()),
-                invalidate_coherent_buffers: RefCell::new(Vec::new()),
+                coherent_buffers: Mutex::new(CoherentBuffers {
```

Comment:
note: I tend to abuse `derive(Default)` for cases like this, but I don't actually recommend it